hs_core/hs_rdf.py

```diff
@@ -0,0 +1,358 @@
+        """Default implementation that ingests by convention"""
+        term = cls.get_class_term()
+        if not term:
+            metadata_nodes = [subject]
+        else:
+            metadata_nodes = graph.objects(subject=subject, predicate=term)
+        for metadata_node in metadata_nodes:
+            value_dict = {}
+            for field in cls._meta.fields:
+                if cls.ignored_fields and field.name in cls.ignored_fields:
+                    continue
+                field_term = cls.get_field_term(field.name)
+                val = graph.value(metadata_node, field_term)
+                if val is not None:
+                    value_dict[field.name] = str(val)
+            if value_dict:
+                cls.create(content_object=content_object, **value_dict)
+
+
+def rdf_terms(class_term, **field_terms):
+    """
+    Deocorator for mapping a class and fields to RDF terms.  The first parameter maps the class to your specified
+    term.  field_terms are key/values with the key matching a class's field and the value being the mapped term.
+
+    :raises ValidationError when a field_term key does not match a field on the class"""
+    def decorator(obj):
+        obj.rdf_term = class_term
+        for k, v in field_terms.items():
+            if not hasattr(obj, k):
+                raise ValidationError("field {} not found".format(k))
+            setattr(obj, k + '_rdf_term', v)
+        return obj
+    return decorator
+
+
+class HydroPrettyXMLSerializer(Serializer):
+    """Same as PrettyXMLSerializer but with stripped out node ids"""
+
+    def __init__(self, store, max_depth=3):
+        super(HydroPrettyXMLSerializer, self).__init__(store)
+        self.forceRDFAbout = set()
+
+    def serialize(self, stream, base=None, encoding=None, **args):
+        self.__serialized = {}
+        store = self.store
+        self.base = base
+        self.max_depth = args.get("max_depth", 3)
+        assert self.max_depth > 0, "max_depth must be greater than 0"
+
+        self.nm = nm = store.namespace_manager
+        self.writer = writer = XMLWriter(stream, nm, encoding)
+        namespaces = {}
+
+        possible = set(store.predicates()).union(
+            store.objects(None, RDF.type))
+
+        for predicate in possible:
+            prefix, namespace, local = nm.compute_qname(predicate)
+            namespaces[prefix] = namespace
+
+        namespaces["rdf"] = "http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+
+        writer.push(RDF.RDF)
+
+        if "xml_base" in args:
+            writer.attribute(XMLBASE, args["xml_base"])
+
+        writer.namespaces(iter(list(namespaces.items())))
+
+        # Write out subjects that can not be inline
+        for subject in store.subjects():
+            if (None, None, subject) in store:
+                if (subject, None, subject) in store:
+                    self.subject(subject, 1)
+            else:
+                self.subject(subject, 1)
```

Comment:
This block of if..if..else logic is confusing to me. Perhaps worth adding a bit of more comments.